### PR TITLE
#166841007 User signin endpoint

### PR DESCRIPTION
**Pull request by chokonaira**
**What does this PR do?**

-An endpoint for existing users to sign in on Authors Heaven

**Description of Task to be completed?**

-Integrated JWT for authentication - user sign-in route

**How should this be manually tested?**

-git clone https://github.com/andela/ah-nyati-backend.git
-navigate to http://localhost:3000/api/v1/auth/login

**Any background context you want to provide?**

N/A

**What are the relevant pivotal tracker stories?**

[#166841007](https://www.pivotaltracker.com/story/show/166841007)

**Screenshots (if appropriate)**

Screenshot 2019-07-06 at 10 48 51 AM

**Questions:**
-N/A

### CODE REVIEW

File: src/controllers/Auth.js [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 import { Auth, hashPassword } from '../helpers/helpers';
-import auth from '../middleware/Auth';
 import { Blacklist, User } from '../db/models';
 import template from '../helpers/mail/mailTemplate/signupEmailTemplate';
 import sendEmail from '../helpers/mail/mailer';
 
 const { generateToken, verifyToken } = Auth;
-
+dotenv.config();
+const isTest = process.env.NODE_ENV === 'test';
```
Comment (fob413) [inline src/controllers/Auth.js]: This should come from the config.js file

File: src/helpers/helpers.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ export class Auth {
   /**
    *
    *@description Verifies token
-   * @param { string } token
+   * @param {*} token
```
Comment (fob413) [inline src/helpers/helpers.js]: please revert this change with what is on staging

File: src/helpers/mail/mailer.js [hunk 3]
```diff
@@ -1,33 +1,14 @@
...
 
-const transport = nodemailer.createTransport(
-  nodemailerSendgrid(
-    {
-      apiKey: sendgrid
-    }
-  )
-);
+dotenv.config();
+
```
Comment (fob413) [inline src/helpers/mail/mailer.js:6]: please revert this change with what is on staging

**Review by phembarl — APPROVED** [review]

**Review by czarjulius — APPROVED** [review]

**Review by codeBlock-1984 — APPROVED** [review]
Nice job @chokonaira.
